jujugui/static/gui/src/app/components/deployment-summary/test-deployment-summary.js

```diff
@@ -101,12 +101,12 @@ describe('DeploymentSummary', function() {
           </div>
           <div className="deployment-summary__footer">
             <juju.components.GenericButton
-              type="clear-changes"
+              type="inline-neutral"
               action={summaryClearAction}
               title="Clear changes" />
             <juju.components.GenericButton
               action={deployButtonAction}
-              type="confirm"
+              type="inline-positive"
```

Comment:
The inline makes them `display: inline-block` and most importantly there width is auto so dynamic to the content. But in half the places we want this behaviour and the other half we want the width to be the width of its parent container. 

So, I settled on a syntax of `button--position` for a green full width button and `button--inline-positive` for a green button that only takes the space of the content. 

I know its a little messy but I went through a number of solutions before I settled on this one for its flexibility and easy of changing a button from full width to inline. This also means we dont have lots of bespoke styling for component buttons like:
```css
.inspector [class^="button--"] { 
  width: 100%;
  display: block;
}
```